enigma-principal/app/src/controller/km_http_server.rs

```diff
@@ -1,184 +1,167 @@
-    }
-}
-
-impl TryInto<U256> for StringWrapper {
-    type Error = Error;
-
-    fn try_into(self) -> Result<U256, Self::Error> {
-        let result = U256::from_dec_str(&self.0);
-        if let Ok(v) = result {
-            Ok(v)
-        } else {
-            Err(RequestValueErr {
-                request: METHOD_GET_STATE_KEYS.to_string(),
-                message: "Cannot create a number.".to_string(),
-            }.into())
-        }
-    }
-}
-
 #[derive(Deserialize, Debug, Clone)]
 pub struct StateKeyRequest {
-    pub data: StringWrapper,
-    pub sig: StringWrapper,
-    pub block_number: Option<StringWrapper>,
-    pub addresses: Option<Vec<String>>,
+    #[serde(with = "hex_serde")]
+    data: Vec<u8>,
+    #[serde(with = "hex_serde")]
+    sig: Vec<u8>,
+    block_number: Option<U256>,
+    addresses: Option<Vec<Hash256>>,
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct StateKeyResponse {
-    pub data: StringWrapper,
-    pub sig: StringWrapper,
+    #[serde(with = "hex_serde")]
+    data: Vec<u8>,
+    #[serde(with = "hex_serde")]
+    sig: Vec<u8>,
 }
 
-impl<H: ToHex> From<H> for StringWrapper {
-    fn from(bytes: H) -> Self { StringWrapper(bytes.to_hex()) }
+impl StateKeyResponse {
+    pub fn new(data: Vec<u8>, sig: Vec<u8>) -> Self {
+        StateKeyResponse{data, sig}
+    }
 }
 
-pub struct PrincipalHttpServer {
-    epoch_provider: Arc<EpochProvider>,
+pub struct KMHttpServer {
+    controller: Arc<KMController>,
     pub port: u16,
 }
 
 impl StateKeyRequest {
-    pub fn get_data(&self) -> Result<Vec<u8>, Error> { Ok(self.data.0.from_hex()?) }
+    pub fn new(data: Vec<u8>, sig: Vec<u8>, block_number: Option<U256>, addresses: Option<Vec<Hash256>>) -> Self {
+        StateKeyRequest{data, sig, block_number, addresses}
+    }
 
-    pub fn get_sig(&self) -> Result<[u8; 65], Error> {
-        self.sig.clone().try_into()
+    pub fn get_data(&self) -> Vec<u8> { self.data.clone() }
+
+    pub fn get_sig(&self) -> Result<[u8; 65], HTTPServerError> {
+        let mut sig = [0u8; 65];
+        if sig.len() != self.sig.len() {
+            return Err(HTTPServerError::SigErr(self.sig.len()))
+        }
+        sig.copy_from_slice(&self.sig);
+        Ok(sig)
     }
 }
 
-impl PrincipalHttpServer {
-    pub fn new(epoch_provider: Arc<EpochProvider>, port: u16) -> PrincipalHttpServer { PrincipalHttpServer { epoch_provider, port } }
+impl KMHttpServer {
+    pub fn new(controller: Arc<KMController>, port: u16) -> KMHttpServer { KMHttpServer { controller, port } }
 
     #[logfn(DEBUG)]
-    fn find_epoch_contract_addresses(request: &StateKeyRequest, msg: &PrincipalMessage, epoch_state: &EpochState) -> Result<Vec<ContractAddress>, Error> {
-        let image = msg.to_sign()?;
+    fn find_epoch_contract_addresses(request: &StateKeyRequest, msg: &PrincipalMessage, epoch_state: &SignedEpoch) -> Result<Vec<Hash256>, HTTPServerError> {
+        let image = msg.to_sign().map_err(|e| HTTPServerError::MessagingErr)?;
```

Comment:
```suggestion
        let image = msg.to_sign().or(HTTPServerError::MessagingErr)?;
```
and same in the rest of the module